<?xml version="1.0" ?>
<sdf version="1.9">
  <world name="city_1_world">
    <physics type="ode">
      <max_step_size>0.002</max_step_size>
      <real_time_factor>1.0</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>
    <plugin name='gz::sim::systems::Physics' filename='libgz-sim-physics-system.so'/>
    <plugin name='gz::sim::systems::UserCommands' filename='libgz-sim-user-commands-system.so'/>
    <plugin name='gz::sim::systems::SceneBroadcaster' filename='libgz-sim-scene-broadcaster-system.so'/>
    <plugin name='gz::sim::systems::Contact' filename='libgz-sim-contact-system.so'/>
    <plugin name='gz::sim::systems::Imu' filename='libgz-sim-imu-system.so'/>
    <plugin name='gz::sim::systems::AirPressure' filename='libgz-sim-air-pressure-system.so'/>
    <plugin name='gz::sim::systems::Sensors' filename='libgz-sim-sensors-system.so'>
      <render_engine>ogre2</render_engine>
    </plugin>
    <gravity>0 0 -9.80665</gravity>
    <magnetic_field>6e-06 2.3e-05 -4.2e-05</magnetic_field>
    <atmosphere type='adiabatic'/>
    <scene>
      <grid>false</grid>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>false</shadows>
    </scene>
    <model name="ground_plane">
      <pose>60 60 0 0 0 0</pose>
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>150 150</size>
            </plane>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>150 150</size>
            </plane>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
      </link>
    </model>
    <include>
      <pose>0 0 0 0 0 0</pose>
      <uri>model://world_simple</uri>
    </include>
    <include>
        <pose>42.506 22.338 2.0109 1.5708 0 4.71238</pose>
        <uri>model://aruco_markers/aruco_marker_1</uri>
        </include>
    <include>
        <pose>25.233 73.555 10.522 1.5708 0 0.807</pose>
        <uri>model://aruco_markers/aruco_marker_2</uri>
        </include>
    <include>
        <pose>45.295 103.64 20.87 1.5708 0 1.8052</pose>
        <uri>model://aruco_markers/aruco_marker_3</uri>
        </include>
    <include>
        <pose>83.253 87.747 0.72931 1.5708 0 3.8275</pose>
        <uri>model://aruco_markers/aruco_marker_4</uri>
        </include>
    <include>
        <pose>63.303 12.454 20.22 1.5708 0 5.8486</pose>
        <uri>model://aruco_markers/aruco_marker_5</uri>
        </include>
    <include>
        <pose>48.633 112.05 11.207 1.5708 0 2.42</pose>
        <uri>model://aruco_markers/aruco_marker_6</uri>
        </include>
    <include>
        <pose>70.576 112.46 2.0234 1.5708 0 3.944</pose>
        <uri>model://aruco_markers/aruco_marker_7</uri>
        </include>
    <include>
        <pose>47.351 62.635 15.552 1.5708 0 2.6396</pose>
        <uri>model://aruco_markers/aruco_marker_8</uri>
        </include>
    <include>
        <pose>20.048 78.774 6.9019 1.5708 0 3.9954</pose>
        <uri>model://aruco_markers/aruco_marker_9</uri>
        </include>
    <include>
        <pose>22.823 80.502 2.7143 1.5708 0 3.1512</pose>
        <uri>model://aruco_markers/aruco_marker_10</uri>
        </include>
    <light type="directional" name="sun">
      <cast_shadows>false</cast_shadows>
      <pose>0 0 100 0 0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>
  </world>
</sdf>
